kratos/includes/model_part.h

```diff
@@ -1457,193 +1498,140 @@ class KRATOS_API(KRATOS_CORE) ModelPart final
     GeometryType::Pointer CreateNewGeometry(
         const std::string& rGeometryTypeName,
         const std::string& rGeometryIdentifierName,
-        GeometryType::Pointer pGeometry
+        GeometryType::Pointer pGeometry,
+        IndexType ThisIndex = 0
         );
 
-    /// Adds a geometry to the geometry container.
-    void AddGeometry(typename GeometryType::Pointer pNewGeometry);
-
-    /// Inserts a list of geometries to a submodelpart provided their Id. Does nothing if applied to the top model part
-    void AddGeometries(std::vector<IndexType> const& GeometriesIds);
-
-    /// Inserts a list of geometries to a submodelpart provided their iterators
-    template<class TIteratorType >
-    void AddGeometries(TIteratorType GeometryBegin,  TIteratorType GeometriesEnd, IndexType ThisIndex = 0)
-    {
-        KRATOS_TRY
-
-        ModelPart* p_root_model_part = &this->GetRootModelPart();
-
-        block_for_each(GeometryBegin, GeometriesEnd, [p_root_model_part](const auto& prGeometry) {
-            const auto& r_geometry = ReferenceGetter<GeometryContainerType::value_type>::Get(prGeometry);
-            const auto& r_geometries = p_root_model_part->Geometries();
-            auto it_found = r_geometries.find(r_geometry.Id());
-            if (it_found != p_root_model_part->GeometriesEnd()) {
-                if (GeometryType::HasSameGeometryType(r_geometry, *it_found)) { // Check the geometry type and connectivities
-                    for (IndexType i_pt = 0; i_pt < r_geometry.PointsNumber(); ++i_pt) {
-                        KRATOS_ERROR_IF((r_geometry)[i_pt].Id() != (*it_found)[i_pt].Id()) << "Attempting to add a new geometry with Id: " << r_geometry.Id() << ". A same type geometry with same Id but different connectivities already exists." << std::endl;
-                    }
-                } else if(&(*it_found) != &r_geometry) { // Check if the pointee coincides
-                    KRATOS_ERROR << "Attempting to add a new geometry with Id: " << it_found->Id() << ". A different geometry with the same Id already exists." << std::endl;
-                }
-            }
-        });
-
-        InsertEntityRange([](ModelPart* pModelPart) {
-            return &(pModelPart->Geometries());
-        }, GeometryBegin, GeometriesEnd);
-
-        KRATOS_CATCH("")
+    /** Returns if the Element corresponding to it's identifier exists */
+    bool HasGeometry(IndexType GeometryId, IndexType ThisIndex = 0) const {
+        return GetMesh(ThisIndex).HasGeometry(GeometryId);
```

Comment:
```suggestion
    bool HasGeometry(IndexType GeometryId) const
    {
        return GetMesh(0).HasGeometry(GeometryId);
```
Also note the stylish suggestion to keep consistency with the rest of the file.